### ISSUE TITLE
Benchmark performance and compare it to baseline

### ISSUE THREAD

**Issue by cjappl** [realtime-sanitizer/radsan#16]
From some third party feedback:

> The perf is absolutely abysmal, and I don't see any evidence in spldaemon.log that the scene (Rainy Cabin) was loaded. I also observe that hitting monitormin.json basically hangs as does getstatuses. So I'm not sure how far I'm going to get tickling
the sanitizer.

This translates to "Performance appeared to be so bad that I couldn't even use the app as I wanted to".

To get to the bottom of this, we should do a little A/B profiling to see how bad running with `-fsanitize=realtime` is, and if it shows considerable issue, fix it. 

I propose something along the lines of:

1. Compile a program with gbench with and without -fsanitize=realtime, see how they compare
2. Compile a bigger program with some defined execution time with and without the flag, see how they compare.

TL;DR deliverables:
1. Benchmark performance of using the realtime sanitizer
2. Fix any major problems with performance
3. Document the runtime performance cost of running with the sanitizer so users know what it is

**Comment by cjappl** [comment]
Some factors that come to mind may be:

* Number of function calls, or nested function calls
* How deep the trace is below your [[clang::realtime]] function

**Comment by cjappl** [comment]
Some good-ish news is that the reporting codebase also has issues running tsan, and behaves in a very similar way. This could indicate a problem with that codebase, and not with RADSAN proper.

I still think this investigation is worth doing, but at least it doesn't appear that we are doing something super nasty.

**Comment by cjappl** [comment]
I think we have a reasonably good idea of what the performance penalty of rtsan is -- very negligible. I am going to close this issue for now and we can re-open it later if it becomes a sticking point
